components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/PhenotipsFamily.java

```diff
@@ -133,22 +119,16 @@ public DocumentReference getDocumentReference()
     public List<Patient> getMembers()
     {
         List<String> memberIds = this.getMembersIds();
-        List<Patient> members = new ArrayList<>(memberIds.size());
-        for (String memberId : memberIds) {
-            Patient patient = PhenotipsFamily.patientRepository.get(memberId);
-            members.add(patient);
-        }
-        return members;
+        return memberIds.stream()
+            .map(memberId -> PhenotipsFamily.patientRepository.get(memberId))
+            .collect(Collectors.toCollection(() -> new ArrayList<>(memberIds.size())));
```

Comment:
This probably should be a subject of a separate discussion, but just curious why rewrite an already written `for` loop in favor of streams - from what I read about streams, the main benefit is ability to parallelize them with `parallel()`, which is not very relevant here, otherwise old school `for` loops usually perform faster, put less pressure on garbage collector due to less wrapper objects, and are easier to debug (this is why I guess there used to be an unnecessary temporary variable `patient` in the original code), and the code is about equally easy to read in both cases. Streams and maps and lambdas are useful/cool, but why use them in such a simple case?

Reply:
@allasm my guess would be: fewer lines of code :)